lib/libzrepl/mgmt_conn.c

```diff
@@ -1402,6 +1407,13 @@ process_message(uzfs_mgmt_conn_t *conn)
 		}
 		if (uzfs_zvol_get_status(zinfo->main_zv) !=
 		    ZVOL_STATUS_HEALTHY) {
+			if (hdrp->opcode == ZVOL_OPCODE_SNAP_CREATE &&
+			    ZVOL_IS_REBUILDING_AFS(zinfo->main_zv)) {
+				mutex_enter(&zinfo->main_zv->rebuild_mtx);
```

Comment:
shall we have a print here, before taking the lock?

Reply:
sure